cmd/avalanche.go

```diff
@@ -147,18 +147,19 @@ func main() {
 			}()
 		}
 
-		// First cut: just send the metrics once then exit
-		if err := metrics.SendRemoteWrite(config, reg); err != nil {
-			log.Fatal(err)
-		}
-		if *remotePprofInterval > 0 {
-			done <- struct{}{}
-			wg.Wait()
-		}
-		return
+		ctx, cancel := context.WithCancel(context.Background())
+		g.Add(func() error {
+			if err := metrics.SendRemoteWrite(ctx, config, reg); err != nil {
+				log.Fatal(err)
```

Comment:
it feels like it should pass error to parent and then we can fatal there, that would at least cancel the context, not sure if imporatnt though

Reply:
ups, yup!

Reply:
Fatal is called by g.Run already even, so just passing error is enough